content/chapter-destinations/destination-s3/_index.md

```diff
@@ -0,0 +1,205 @@
+
+## bucket()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The name of the S3 bucket, for example, `my-bucket`
+
+## canned-acl()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `empty` |
+
+*Description:* The ACL assigned to the object, if specified, for example, `bucket-owner-read`. The following values are valid:
+
+`authenticated-read`, `aws-exec-read`, `bucket-owner-full-control`, `bucket-owner-read`, `log-delivery-write`, `private`, `public-read`, `public-read-write`
+
+If you configure an invalid value, the default is used.
+
+## chunk-size()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `5MiB` |
+
+*Description:* The size of log messages that  {{% param "product_name" %}} writes to the S3 object in a batch.
+<!-- FIXME how does is relate to compression? -->
+
+## compression()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | boolean |
+| Default: | `no` |
+
+*Description:* <!-- FIXME -->
+
+## compresslevel()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer |
+| Default: | `9` |
+
+*Description:* <!-- FIXME -->
+
+## flush-grace-period()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer (seconds) |
+| Default: | `60` |
+
+*Description:* After the grace period expires and no new messages are routed to the destination, {{% param "product_name" %}} flushes the contents of the buffer to the S3 object even if the volume of the messages in the buffer is lower than [`chunk-size()`](#chunk-size).
+
+## max-object-size()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `5120GiB` |
+
+*Description:* The maximal size of the S3 object.
+<!-- What happens if we reach it, we start a new object? -->
+
+## max-pending-uploads()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer |
+| Default: | `32` |
+
+*Description:* <!-- FIXME -->
+
+## object-key()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The [object key](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html) (or key name), which uniquely identifies the object in an Amazon S3 bucket.
+
+## object-key-timestamp()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The timestamp added to the object.
+<!-- FIXME ?? -->
```

Comment:
See: https://github.com/syslog-ng/syslog-ng/releases/tag/syslog-ng-4.4.0

> Rotation based on timestamp
The object-key-timestamp() option can be used to set a datetime related template, which gets
appended to the end of the object (e.g. "${R_MONTH_ABBREV}${R_DAY}" => "-Sep25"). When a log
message arrives with a newer timestamp template resolution, the previous timestamped object gets
finised and a new one is started with the new timestamp. Backfill messages do not reopen and append
the old object, but starts a new object with the key having an index appended to the old object.